src/parser/utils.rs

```diff
@@ -19,11 +19,20 @@ use crate::{
     term::{
         make as mk_term,
         record::{Field, FieldMetadata, RecordAttrs, RecordData},
-        BinaryOp, LabeledType, LetMetadata, RichTerm, StrChunk, Term, TypeAnnotation, UnaryOp,
+        BinaryOp, LabeledType, LetMetadata, Rational, RichTerm, StrChunk, Term, TypeAnnotation,
+        UnaryOp,
     },
     types::{TypeF, Types},
 };
 
+use malachite::num::conversion::traits::FromSciString;
+
+pub struct ParseNumberError;
```

Comment:
Would it be useful if `ParseNumberError` stored the string that failed to parse?

Reply:
I think it is customary to not store anything inside parsing errors in Rust, because the caller to the `parse()` function surely still has access to the original `&str`. So unless we add more precise information, like an error message, or the incriminated position inside the string, just returning back the original argument is not really useful.